superset/databases/commands/test_connection.py

```diff
@@ -60,20 +61,34 @@ def run(self) -> None:
                 database.db_engine_spec.mutate_db_for_connection_test(database)
                 username = self._actor.username if self._actor is not None else None
                 engine = database.get_sqla_engine(user_name=username)
-            with closing(engine.raw_connection()) as conn:
-                if not engine.dialect.do_ping(conn):
-                    raise DBAPIError(None, None, None)
+                with closing(engine.raw_connection()) as conn:
```

Comment:
I'm not sure what the intention was here, but if database is None, then this statement fails because engine is not defined, so I put it inside this block.